src/Lean/Expr.lean

```diff
@@ -919,6 +919,14 @@ private def getAppNumArgsAux : Expr → Nat → Nat
 def getAppNumArgs (e : Expr) : Nat :=
   getAppNumArgsAux e 0
 
+/--
+If the given expression is a sequence of function applications `f a₁ .. aₙ`,
+returns `f a₁ .. aₖ` where `k` is minimal such that `n - k ≤ maxArgs`.
```

Comment:
Should it be `maxRemainingArgs` here for clarity?

Reply:
I find my comment to be confusing. I want `getBoundedAppFn m e` to pop off at most `m` arguments, and I want it to be "dual" to `getBoundedAppArgs` in the sense that `mkAppN (e.getBoundedAppFn m) (e.getBoundedAppArgs m) = e`.

Really, the use case is extracting a function of a particular arity, but I found it easier to think about these functions in their current form, where it is up to the caller to decide how many arguments to pop off.

Reply:
I tried to clarify the docstring.